### PR TITLE
Disable generation of hyperd.log

### LINKED ISSUE #291: Make frames_from_hyper not generate hyperd.log
**Is your feature request related to a problem? Please describe.**
`frames_from_hyper` uses a `HyperProcess` that generates a hyperd.log file. I'd like it not to generate the log files by default, or at least have an option to disable logging.

**Describe the solution you'd like**
Adding `{"log_config": ""}` to `HyperProcess` arguments would do it.

```python
def frames_from_hyper(
    source: Union[str, pathlib.Path],
    return_type: Literal["pandas", "polars", "pyarrow"] = "pandas",
):
    """See api.rst for documentation."""
    result = {}

    table_names = []
    with tempfile.TemporaryDirectory() as tmp_dir, tab_api.HyperProcess(
        tab_api.Telemetry.DO_NOT_SEND_USAGE_DATA_TO_TABLEAU, 
        parameters={"log_config": ""},
    ) as hpe:
        tmp_db = shutil.copy(source, tmp_dir)
        with tab_api.Connection(hpe.endpoint, tmp_db) as connection:
            for schema in connection.catalog.get_schema_names():
                for table in connection.catalog.get_table_names(schema=schema):
                    table_names.append(table)

    for table in table_names:
        result[table] = frame_from_hyper(
            source=source,
            table=table,
            return_type=return_type,
        )

    return result
```

**Describe alternatives you've considered**
`frames_from_hyper` could take a parameter to enable the log file for users that do want it.

**Additional context**
I use pantab 4.1.0.

### PR DESCRIPTION
**Pull request by fxstempfelals**
closes #291

### CODE REVIEW

**Comment by WillAyd** [comment]
Looks great - can you also add a test to `tests/test_reader.py` to make sure this file does not exist after calling frame_from_hyper and frames_from_hyper?

File: newsfragments/291.fix [hunk 1]
```diff
@@ -0,0 +1 @@
+Disables generation of tableauhyperapi's hyperd.log file.
```
Comment (WillAyd) [inline newsfragments/291.fix:1]: Pre-commit is failing due to the lack of a newline at the end of this file

**Comment by fxstempfelals** [comment]
@WillAyd added the tests and the newline. I'm sorry but I have trouble setting up the environment to run the tests and pre-commit. Still hope everything goes fine since it's such a simple case...

**Comment by WillAyd** [comment]
Awesome work - thanks @fxstempfelals

**Comment by WillAyd** [comment]
Looks like this is still happening on main - @fxstempfelals any chance you can take a look?

https://github.com/innobi/pantab/actions/runs/8645626809

**Comment by fxstempfelals** [comment]
@WillAyd I believe this is because HyperProcess is used in other tests' code and they generate the file, not the tested code.
This PR proposes a fix but I still cannot run the tests... https://github.com/innobi/pantab/pull/294

**Comment by WillAyd** [comment]
I think I merged this too soon - looks to be in a failed state. Sorry for overlooking that!

For now going to revert but happy to re-merge this in later if it can get green in CI

**Pull request merged**